docs/content/envoy-tutorial-istio.md

```diff
@@ -0,0 +1,196 @@
+See Istio's [Quick Start](https://istio.io/docs/setup/kubernetes/install/kubernetes/) page to get started.
+
+## Steps
+
+### 1. Install OPA-Envoy
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/open-policy-agent/opa-istio-plugin/master/examples/istio/quick_start.yaml
+```
+
+The `quick_start.yaml` manifest defines the following resources:
+
+* External Authorization Filter to direct authorization checks to the OPA-Envoy sidecar. See `kubectl -n istio-system get envoyfilter ext-authz` for details.
+
+* Kubernetes namespace (`opa-istio`) for OPA-Envoy control plane components.
+
+* Kubernetes admission controller in the `opa-istio` namespace that automatically injects the OPA-Envoy sidecar into pods in namespaces labelled with `opa-istio-injection=enabled`.
+
+* OPA configuration file and an OPA policy into ConfigMaps in the namespace where the app will be deployed, e.g., `default`.
+  The following is the example OPA policy:
+        
+    * alice is granted a **guest** role and can perform a `GET` request to `/productpage`.
+    * bob is granted an **admin** role and can perform a `GET` to `/productpage` and `/api/v1/products`.
+
+    ```live:example:module:openable
+    package istio.authz
+  
+    import input.attributes.request.http as http_request
+    import input.parsed_path
+    
+    default allow = false
+    
+    allow {
+        parsed_path[0] == "health"
+        http_request.method == "GET"
+    }
+    
+    allow {
+        roles_for_user[r]
+        required_roles[r]
+    }
+    
+    roles_for_user[r] {
+        r := user_roles[user_name][_]
+    }
+    
+    required_roles[r] {
+        perm := role_perms[r][_]
+        perm.method = http_request.method
+        perm.path = http_request.path
+    }
+    
+    user_name = parsed {
+        [_, encoded] := split(http_request.headers.authorization, " ")
+        [parsed, _] := split(base64url.decode(encoded), ":")
+    }
+    
+    user_roles = {
+        "alice": ["guest"],
+        "bob": ["admin"]
+    }
+    
+    role_perms = {
+        "guest": [
+            {"method": "GET",  "path": "/productpage"},
+        ],
+        "admin": [
+            {"method": "GET",  "path": "/productpage"},
+            {"method": "GET",  "path": "/api/v1/products"},
+        ],
+    }
+    ```
+
+    OPA is configured to query for the `data.istio.authz.allow`
+    decision. If the response is `true` the operation is allowed, otherwise the
+    operation is denied. Sample input received by OPA is shown below:
+
+    ```live:example:query:hidden
+    data.istio.authz.allow
+    ```
+  
+    ```live:example:input
+    {
+        "attributes": {
+            "request": {
+                "http": {
+                    "method": "GET",
+                    "path": "/productpage",
+                    "headers": {
+                        "authorization": "Bearer YWxpY2U6cGFzc3dvcmQ="
```

Comment:
The policy will work either way, but this should be "Basic ..." rather than "Bearer ...", right?

Reply:
Yes, updated.